eshell-info-banner.el

```diff
@@ -292,20 +292,46 @@ the warning face with a battery level of 25% or less."
                                      (number-to-string percentage)
                                      :inherit (eshell-info-banner--get-color-percentage (- 100.0 percentage)))))))))
 
+                                        ; Operating system identification ;;;;;;;;;;;;;;;;;;
+(defun eshell-info-banner--get-os-information-from-release-file (&optional release-file)
+  "Read the operating system from the given RELEASE-FILE.
+
+If RELEASE-FILE is nil, use '/etc/os-release'."
+  (setq release-file (or release-file "/etc/os-release"))
```

Comment:
Please use `setq-local` instead, we don’t need this variable to be set for the whole system, just in the function, or wrap it in a 
```emacs-lisp
(let ((release-file (or release-file "/etc/os-release")))
  ;; code
  )
```

Reply:
Doesn't `setq` only set the function-local  `release-file` variable, which is a function argument? This shouldn't introduce a global variable:
```lisp
(eshell-info-banner--get-os-information-from-release-file)
(boundp 'release-file) ; nil
```
I can use the `let` variant if necessary though.